src/Compilers/CSharp/Portable/Symbols/DiscardSymbol.cs

```diff
@@ -5,18 +5,21 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
-    internal class DiscardSymbol : Symbol, IDiscardSymbol
+    internal sealed class DiscardSymbol : Symbol, IDiscardSymbol
     {
-        private readonly TypeSymbol _type;
-
         public DiscardSymbol(TypeSymbol type)
         {
             Debug.Assert((object)type != null);
-            _type = type;
+            Type = type;
         }
 
-        ITypeSymbol IDiscardSymbol.Type => _type;
-        public TypeSymbol Type => _type;
+        ITypeSymbol IDiscardSymbol.Type => Type;
+        public TypeSymbol Type { get; }
+
+        /// <summary>
+        /// Produce a fresh discard symbol for testing.
+        /// </summary>
+        internal static DiscardSymbol CreateForTest(ITypeSymbol type) => new DiscardSymbol((TypeSymbol)type);
 
```

Comment:
The entire type is internal, it looks like using constructor would be sufficient.